### DISCUSSION TITLE
Could this parse out SQL variables?

### DISCUSSION THREAD

**Discussion by AStvnMC** [JSQLParser/JSqlParser#1562] [Q&A] [answered]
Hi, 

Two Questions:

First Question
Is the parser able to parse multi-line sql queries at once?
For example, if the query sent was:
DECLARE @var nvarchar(max) = 'Man''s best friend'
"SELECT * FROM tab1; SELECT * FROM tab2"

Second Question
Is the parser able to parse SQL variables?
For example if the query was: DECLARE @var nvarchar(max) = 'Man''s best friend'

thanks!

**Comment by manticore-projects** [comment, marked as answer]
Greetings.

Yes and yes. [You can try it online here](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20CIUQwgMgggSiAEABAbgQwE7wHZvQYwAsN4AKAW1QA8BKeAXngHIBZVLRxgZ3gCMBTTgBd4AM3QBLPlgAmjAFABuOcoDKICOAAq8AFRyAYjADyzeINQ8AjIuVy1GsNr2GTZiwCZFQA+)

Use the `parseStatements(String sqlStr)` instead of the `parse(String sqlStr)` method when parsing multiple statements at once.

**Reply by AStvnMC** [reply]
Hi, I can't seem to get the link to work. It just keeps on loading.

**Comment by manticore-projects** [comment]
Works here. Even on a mobile phone. Although you will need to allow javascript and html5.Sent from my Galaxy
-------- Original message --------From: AStvnMC ***@***.***> Date: 13/06/2022  20:43  (GMT+07:00) To: JSQLParser/JSqlParser ***@***.***> Cc: manticore-projects ***@***.***>, Comment ***@***.***> Subject: Re: [JSQLParser/JSqlParser] Could this parse out SQL variables?
  (Discussion #1562) 
Hi, I can't seem to get the link to work. It just keeps on loading.

**Reply by AStvnMC** [reply]
thanks for your reply its all working now :)

**Reply by AStvnMC** [reply]
If I wanted to parse out the variable name (@var) in this example, and its value 'Man''s best friend'. Is it possible? If so, how would the code to parse it out look like

**Reply by manticore-projects** [reply]
> If I wanted to parse out the variable name (@var) in this example, and its value 'Man''s best friend'. Is it possible? If so, how would the code to parse it out look like

```java
        import net.sf.jsqlparser.JSQLParserException;
        import net.sf.jsqlparser.parser.CCJSqlParserUtil;

         sqlStr="DECLARE @var nvarchar(max) = 'Man''s best friend';\n" +
                "SELECT * FROM tab1; SELECT * FROM tab2;";

        Statements statements = CCJSqlParserUtil.parseStatements(sqlStr);
        DeclareStatement declareStatement = (DeclareStatement) statements.getStatements().get(0);

        System.out.println("User Variable = " + declareStatement.getTypeDefExprList().get(0).userVariable);
        System.out.println("Type = " + declareStatement.getTypeDefExprList().get(0).colDataType);

        System.out.println("Default Value = " + declareStatement.getTypeDefExprList().get(0).defaultExpr);
```


```
User Variable = @var
Type = nvarchar (max)
Default Value = 'Man''s best friend'
```

**Reply by AStvnMC** [reply]
thank you so much! :)
